crates/algorithm/src/lib.rs

```diff
@@ -245,6 +304,9 @@ impl<T: Ord> Sequence for BinaryHeap<T> {
 impl<I: Iterator> Sequence for Peekable<I> {
     type Item = I::Item;
     type Inner = Peekable<I>;
+    fn peek(&mut self) -> Option<&I::Item> {
```

Comment:
If you require the implementation of `peek`, then you can provide a default implementation for `next_if`, or just remove it.